### ISSUE TITLE
Node instances only working with shader material

### ISSUE THREAD

**Issue by frauzufall** [scenerygraphics/scenery#363]
Creating an instance of a node works like this:

```
val box = Box()
val mat = ShaderMaterial.fromFiles("DefaultDeferredInstanced.vert", "DefaultDeferred.frag")
box.material = mat
box.instancedProperties.put("ModelMatrix", { box.model })
scene.addChild(box)

val n1 = Mesh();
n1.instancedProperties.put("ModelMatrix", n1::world);
n1.position = Vector3f(1f, 0.0f, 0.0f)
box.instances.add(n1)

val n2 = Mesh();
n2.instancedProperties.put("ModelMatrix", n2::world);
n2.position = Vector3f(-1f, 0.0f, 0.0f)
box.instances.add(n2)
```

![Screenshot!UNITO-UNDERSCORE!2021-01-11!UNITO-UNDERSCORE!11-15-24](https://user-images.githubusercontent.com/708121/104168832-f14bb000-53fe-11eb-8691-3d7d7f56de6f.png)

.. but not when I set a material for the box like this:

```
val box = Box()
val mat = Material()
box.material = mat
```

![Screenshot!UNITO-UNDERSCORE!2021-01-11!UNITO-UNDERSCORE!11-17-08](https://user-images.githubusercontent.com/708121/104168835-f27cdd00-53fe-11eb-8a56-1e4846afd9f6.png)

It took me a while to figure that out - is this limitation necessary? If yes, then it could maybe be caught if someone adds instances to an object with a non shader based material and at least print a warning to the console.

**Comment by skalarproduktraum** [comment]
Hi @frauzufall! This is actually intended to be working this way, but I can totally see why this has been a source of confusion. Maybe it'd be better to provide a wrapper class for instances that holds references to both the master node and the instances. 

Originally, it was actually intended to work in such a way that the renderer rewrites the shaders if instancing is used. The buffer memory layout is actually different when using instancing, so a different shader is necessary.

**Comment by skalarproduktraum** [comment]
This is fixed with #406 👍
